spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/aot/ConfigClientHints.java

```diff
@@ -0,0 +1,66 @@
+package org.springframework.cloud.config.client.aot;
+
+import org.springframework.aot.hint.MemberCategory;
+import org.springframework.aot.hint.RuntimeHints;
+import org.springframework.aot.hint.RuntimeHintsRegistrar;
+import org.springframework.aot.hint.TypeReference;
+import org.springframework.boot.context.config.ConfigDataLocation;
+import org.springframework.cloud.config.client.ConfigClientAutoConfiguration;
+import org.springframework.cloud.config.client.RetryTemplateFactory;
+import org.springframework.cloud.config.environment.Environment;
+import org.springframework.cloud.config.environment.PropertySource;
+import org.springframework.retry.support.RetryTemplate;
```

Comment:
This is actually from a different Spring project/repo, so optimally the hint should be added there; we can keep it here till it's done, but either add a separate PR there and link here or let me know and I'll get it done.

Reply:
I am going to create a PR at Spring Retry.

Reply:
Just for my understanding. Is best practice to define the hint in the project which defines the class? I asked because to me it looks a bit odd, because Spring Retry does not use RetryTemplate with any kind of reflections, but Spring Cloud Config client does. That's the reason I added the hints here.

Reply:
It's best for them to be available for any other project using that class.